File: app/src/main/java/com/parassidhu/cdlumaths/Pair.java
package com.parassidhu.cdlumaths;

import java.io.File;

public class Pair implements Comparable {
    public long t;
    public File f;

    public Pair(File file) {
        f = file;
        t = file.lastModified();
    }

    public int compareTo(Object o) {
        long u = ((Pair) o).t;
        return t < u ? -1 : t == u ? 0 : 1;
    }
}

File: app/src/main/java/com/parassidhu/cdlumaths/models/Pair.kt
package com.parassidhu.cdlumaths.models

import java.io.File

class Pair(var f: File) : Comparable<Any> {
    var t: Long = 0

    init {
        t = f.lastModified()
    }

    override operator fun compareTo(other: Any): Int {
        val u = (other as Pair).t
        return if (t < u) -1 else if (t == u) 0 else 1
    }
}